From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Wed, 6 Aug 2025 18:06:31 +0200
Subject: cell-broadcast-prompt: Close dialog on swipe
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Fixes: b281cc5d3 ("cell-broadcast-prompt: New dialog for cell broadcast message")
Signed-off-by: Guido Günther <agx@sigxcpu.org>
(cherry picked from commit 17453a8d55b80aa83c09ff6eba67a1408c5f99c9)

Part-of: <https://gitlab.gnome.org/World/Phosh/phosh/-/merge_requests/1771>
(cherry picked from commit 164c5f53933f8033e3b1566d0835d9efd4a06c43)
---
 src/cell-broadcast-prompt.c     | 8 ++++++++
 src/ui/cell-broadcast-prompt.ui | 1 +
 2 files changed, 9 insertions(+)

diff --git a/src/cell-broadcast-prompt.c b/src/cell-broadcast-prompt.c
index d6d0c72..3354ccc 100644
--- a/src/cell-broadcast-prompt.c
+++ b/src/cell-broadcast-prompt.c
@@ -113,6 +113,13 @@ on_ok_clicked (PhoshCellBroadcastPrompt *self)
 }
 
 
+static void
+on_dialog_canceled (PhoshCellBroadcastPrompt *self)
+{
+  g_signal_emit (self, signals[CLOSED], 0);
+}
+
+
 static void
 phosh_cell_broadcast_prompt_finalize (GObject *obj)
 {
@@ -148,6 +155,7 @@ phosh_cell_broadcast_prompt_class_init (PhoshCellBroadcastPromptClass *klass)
   gtk_widget_class_set_template_from_resource (widget_class,
                                                "/mobi/phosh/ui/cell-broadcast-prompt.ui");
   gtk_widget_class_bind_template_child (widget_class, PhoshCellBroadcastPrompt, lbl_msg);
+  gtk_widget_class_bind_template_callback (widget_class, on_dialog_canceled);
   gtk_widget_class_bind_template_callback (widget_class, on_ok_clicked);
 
   gtk_widget_class_set_css_name (widget_class, "phosh-cell-broadcast-prompt");
diff --git a/src/ui/cell-broadcast-prompt.ui b/src/ui/cell-broadcast-prompt.ui
index 1c89d72..4ddfe02 100644
--- a/src/ui/cell-broadcast-prompt.ui
+++ b/src/ui/cell-broadcast-prompt.ui
@@ -2,6 +2,7 @@
 <interface>
   <requires lib="gtk+" version="3.24"/>
   <template class="PhoshCellBroadcastPrompt" parent="PhoshSystemModalDialog">
+    <signal name="dialog-canceled" handler="on_dialog_canceled"/>
     <child type="phosh-dialog-content">
       <object class="GtkBox">
         <property name="visible">1</property>
